### ISSUE TITLE
FEProblemBase::getPostprocessorValueByName should check the availability of the pp first

### ISSUE THREAD

**Issue by YaqiWang** [idaholab/moose#23356]
## Reason
Currently, this function directly relay the call to reporter data. When a postprocessor is not available, it will issue an error like
```
Reporter name "xxx/value" with type "double" is not declared.
```
This error message can give users hard time on what is happening.

## Design
Before relaying the call, we should check the availability of the postprocessor and issue a more straight-forward error like
```
Postprocessor name "xxx" is not available.
```

## Impact
Minor user interface improvement.

**Comment by GiudGiud** [comment]
Very much agree. We should never rely on ReporterInterface behavior if the user has not declared Reporters

**Comment by loganharbour** [comment]
> We should never rely on ReporterInterface behavior if the user has not declared Reporters

I disagree. This interface is very well developed with a producer/consumer relationship. Special case the error? Sure. But we should not duplicate capability.
